prov/verbs/src/verbs_domain.c

```diff
@@ -300,7 +323,37 @@ fi_ibv_domain(struct fid_fabric *fabric, struct fi_info *info,
 	_domain->util_domain.domain_fid.fid.fclass = FI_CLASS_DOMAIN;
 	_domain->util_domain.domain_fid.fid.context = context;
 	_domain->util_domain.domain_fid.fid.ops = &fi_ibv_fid_ops;
-	_domain->util_domain.domain_fid.mr = &fi_ibv_domain_mr_ops;
+
+	fi_ibv_domain_process_exp(_domain);
+
+	if (fi_ibv_gl_data.mr_cache_enable) {
+		_domain->monitor.subscribe = fi_ibv_monitor_subscribe;
+		_domain->monitor.unsubscribe = fi_ibv_monitor_unsubscribe;
+		_domain->monitor.get_event = fi_ibv_monitor_get_event;
+		ofi_monitor_init(&_domain->monitor);
+
+		_domain->cache.size = fi_ibv_gl_data.mr_cache_size;
+		_domain->cache.entry_data_size = sizeof(struct fi_ibv_mem_desc);
+		_domain->cache.add_region = fi_ibv_mr_cache_entry_reg;
+		_domain->cache.delete_region = fi_ibv_mr_cache_entry_dereg;
+		ret = ofi_mr_cache_init(&_domain->util_domain, &_domain->monitor,
+					&_domain->cache);
+		if (ret)
+			goto err4;
```

Comment:
ofi_monitor_init just initializes an atomic variable. If there isn't a future plan to extend the function to do something more, going to err4 is probably fine here, otherwise we may need to call ofi_monitor_cleanup before that.